camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelRunner.java

```diff
@@ -508,20 +531,70 @@ protected void awaitStartup() throws Exception {
      */
     protected RouteBuilder loadRouteFromText(String specification, String specificationType) throws Exception {
         log.debug("Loading route from {} specificationType", specificationType);
-        if (log.isDebugEnabled()) {
+        if (log.isTraceEnabled()) {
             // Dump route spec out here only under trace.  Camel may have substituted key values and we don't want
             // these in the log files
-            log.debug("Loading route (specificationType {}):\n{}", specificationType, specification);
+            log.trace("Loading route (specificationType {}):\n{}", specificationType, specification);
         }
-        ExtendedCamelContext extendedCamelContext = camelContext.adapt(ExtendedCamelContext.class);
-        RoutesLoader loader = extendedCamelContext.getRoutesLoader();
-        Resource resource = ResourceHelper.fromString("in-memory." + specificationType, specification);
-        loader.loadRoutes(resource);
-        RoutesBuilderLoader rbl = loader.getRoutesLoader(specificationType);
-        RoutesBuilder rb = rbl.loadRoutesBuilder(resource);
-        log.debug("Route builder: {}", rb.toString());
+        // First, construct a simple classloader.  This is based on the dependencies defined in the Kamelet. Later,
+        // these will be augmented via discovery since these declarations need not be complete. These are used to
+        // satisfy camel's need to instantiate as part of setting up the routes.
+        Map<String, Set<String>> discResults = new HashMap<>();
+        discResults.put(CamelDiscovery.COMPONENTS_TO_LOAD, Set.of());
+        discResults.put(CamelDiscovery.DATAFORMATS_TO_LOAD, Set.of());
+        ClassLoader cl = buildClassloader(null, discResults);
         
-        return (RouteBuilder) rb;
```

Comment:
instantiating a route from a template is a rather involved process.  That's what all this stuff is.